bridges/NineGagBridge.php

```diff
@@ -0,0 +1,330 @@
+					'Fan Art' => 'animefanart',
+					'Food & Drinks' => 'food',
+					'Football' => 'football',
+					'Fortnite' => 'fortnite',
+					'Funny' => 'funny',
+					'GIF' => 'gif',
+					'Gaming' => 'gaming',
+					'Girl' => 'girl',
+					'Girly Things' => 'girly',
+					'Guy' => 'guy',
+					'History' => 'history',
+					'Home Design' => 'home',
+					'Horror' => 'horror',
+					'K-Pop' => 'kpop',
+					'LEGO' => 'lego',
+					'League of Legends' => 'leagueoflegends',
+					'Movie & TV' => 'movie-tv',
+					'Music' => 'music',
+					'NFK - Not For Kids' => 'nsfw',
+					'Overwatch' => 'overwatch',
+					'PC Master Race' => 'pcmr',
+					'PUBG' => 'pubg',
+					'Pic Of The Day' => 'photography',
+					'Pokémon' => 'pokemon',
+					'Politics' => 'politics',
+					'Relationship' => 'relationship',
+					'Roast Me' => 'roastme',
+					'Satisfying' => 'satisfying',
+					'Savage' => 'savage',
+					'School' => 'school',
+					'Sci-Tech' => 'science',
+					'Sport' => 'sport',
+					'Star Wars' => 'starwars',
+					'Superhero' => 'superhero',
+					'Surreal Memes' => 'surrealmemes',
+					'Timely' => 'timely',
+					'Travel' => 'travel',
+					'Video' => 'video',
+					'WTF' => 'wtf',
+					'Wallpaper' => 'wallpaper',
+					'Warhammer' => 'warhammer',
+				),
+			),
+			't' => array(
+				'name' => 'Type',
+				'type' => 'list',
+				'required' => true,
+				'values' => array(
+					'Hot' => 'hot',
+					'Fresh' => 'fresh',
+				),
+			),
+			'p' => array(
+				'name' => 'Pages',
+				'type' => 'number',
+				'defaultValue' => 3,
+			),
+		),
+	);
+
+	const MIN_NBR_PAGE = 1;
+	const MAX_NBR_PAGE = 6;
+
+	protected $p = null;
+
+	public function collectData() {
+		$url = sprintf(
+			'%sv1/group-posts/group/%s/type/%s?',
+			self::URI,
+			$this->getGroup(),
+			$this->getType()
+		);
+		$cursor = 'c=10';
+		$posts = array();
+		for ($i = 0; $i < $this->getPages(); ++$i) {
+			$content = getContents($url.$cursor);
+			$json = json_decode($content, true);
+			$posts = array_merge($posts, $json['data']['posts']);
+			$cursor = $json['data']['nextCursor'];
+		}
+
+		foreach ($posts as $post) {
+			$item['uri'] = $post['url'];
+			$item['title'] = $post['title'];
+			$item['content'] = self::getContent($post);
+			$item['categories'] = self::getCategories($post);
+			$item['timestamp'] = self::getTimestamp($post);
+
+			$this->items[] = $item;
+		}
+	}
+
+	public function getName() {
+		if ($_GET['bridge'] === substr(get_class($this), 0, -6)) {
+			if ($this->getInput('d')) {
+				$name = $this->getParameterKey('d');
+			} else {
+				$name = sprintf(
+					'%s [%s]',
+					$this->getParameterKey('g'),
+					$this->getParameterKey('t')
+				);
+			}
+
+			return sprintf('%s - %s', '9GAG', $name);
+		}
+
+		return self::NAME;
+	}
+
+	public function getURI() {
+		$uri = $this->getInput('g');
+		if ('default' === $uri) {
+			$uri = $this->getInput('t');
+		}
+
+		return self::URI.$uri;
+	}
+
+	public function getIcon() {
```

Comment:
You don't need to implement the function, RSS-Bridge will link by default to ```http://9gag.com/favicon.ico```

Reply:
Thanks, I didn't know this